src/sentry/static/sentry/app/views/organizationIntegrations/integrationUtil.tsx

```diff
@@ -0,0 +1,37 @@
+import {SentryApp, IntegrationProvider, PluginWithProjectList} from 'app/types';
+
+type AppOrProviderOrPlugin = SentryApp | IntegrationProvider | PluginWithProjectList;
+
+const pluginMapping = {
+  slug: 'id',
+  name: 'name',
+};
+
+const providerMapping = {
+  slug: 'key',
+  name: 'name',
+};
+
+const sentryAppMapping = {
+  slug: 'slug',
+  name: 'name',
+};
+
+type integrationTypes = 'plugin' | 'sentry-app' | 'provider';
+
+const getMapping = (type: integrationTypes) => {
+  switch (type) {
+    case 'plugin':
+      return pluginMapping;
+    case 'sentry-app':
+      return sentryAppMapping;
+    case 'provider':
+      return providerMapping;
+    default:
+      return {};
+  }
+};
+
+export const mapped = (integration: AppOrProviderOrPlugin) => (
```

Comment:
@NisanthanNanthakumar Now that I merged the change where the integration provider has a `slug`, this mapping function is unnecessary. You can always use `slug` and `name`.

Reply:
@scefali you don't think we might need this function for future work?

Reply:
@NisanthanNanthakumar At this point, I'm not sure. The problem is that some of the information we need from integrations are not always stored on the objects themselves such as the installation status or the features. So of the things that we need from the actual integration object themselves, I'm not sure how many uses cases are left where this abstraction makes sense.